src/graph/OrderByExecutor.cpp

```diff
@@ -0,0 +1,226 @@
+        }
+        case Type::double_precision:
+        {
+            if (!(lhs.value_.double_precision == rhs.value_.double_precision)) {
+                return lhs.value_.double_precision < rhs.value_.double_precision;
+            }
+            break;
+        }
+        case Type::str:
+        {
+            if (!(lhs.value_.str == rhs.value_.str)) {
+                return lhs.value_.str < rhs.value_.str;
+            }
+            break;
+        }
+        case Type::timestamp:
+        {
+            if (!(lhs.value_.timestamp == rhs.value_.timestamp)) {
+                return lhs.value_.timestamp < rhs.value_.timestamp;
+            }
+            break;
+        }
+        case Type::year:
+        {
+            if (!(lhs.value_.year == rhs.value_.year)) {
+                return lhs.value_.year < rhs.value_.year;
+            }
+            break;
+        }
+        case Type::month:
+        {
+            if (!(lhs.value_.month == rhs.value_.month)) {
+                return lhs.value_.month < rhs.value_.month;
+            }
+            break;
+        }
+        case Type::date:
+        {
+            if (!(lhs.value_.date == rhs.value_.date)) {
+                return lhs.value_.date < rhs.value_.date;
+            }
+            break;
+        }
+        case Type::datetime:
+        {
+            if (!(lhs.value_.datetime == rhs.value_.datetime)) {
+                return lhs.value_.datetime < rhs.value_.datetime;
+            }
+            break;
+        }
+        default:
+        {
+            return false;
+        }
+    }
+    return false;
+}
+}  // namespace cpp2
+
+OrderByExecutor::OrderByExecutor(Sentence *sentence, ExecutionContext *ectx)
+    : TraverseExecutor(ectx) {
+    sentence_ = static_cast<OrderBySentence*>(sentence);
+}
+
+Status OrderByExecutor::prepare() {
+    return Status::OK();
+}
+
+void OrderByExecutor::feedResult(std::unique_ptr<InterimResult> result) {
+    if (result == nullptr) {
+        return;
+    }
+    DCHECK(sentence_ != nullptr);
+    inputs_ = std::move(result);
+    rows_ = inputs_->getRows();
+
+    auto schema = inputs_->schema();
+    auto factors = sentence_->factors();
+    sortFactors_.reserve(factors.size());
+    for (auto &factor : factors) {
+        auto expr = static_cast<InputPropertyExpression*>(factor->expr());
+        folly::StringPiece field = *(expr->prop());
+        auto fieldIndex = schema->getFieldIndex(field);
+        if (fieldIndex == -1) {
+            LOG(INFO) << "Field(" << field << ") not exist in input schema.";
+            continue;
+        }
+        auto pair = std::make_pair(schema->getFieldIndex(field), factor->orderType());
+        sortFactors_.emplace_back(std::move(pair));
+    }
+}
+
+void OrderByExecutor::execute() {
+    FLOG_INFO("Executing Order By: %s", sentence_->toString().c_str());
+    auto comparator = [this] (cpp2::RowValue& lhs, cpp2::RowValue& rhs) {
+        auto lhsColumns = lhs.get_columns();
```

Comment:
You are copying the columns. 

IMO. You could eliminate the copying with `const auto &lhsColumns = ...`, provided that `lhs` and `rhs` are passed by const reference.

Reply:
thank you!!!